cli/lib/snapshots.ts

```diff
@@ -114,3 +125,64 @@ export async function deleteSnapshotFromAppdata( snapshotUrl: string ): Promise<
 	userData.snapshots.splice( snapshotIndex, 1 );
 	await saveAppdata( userData );
 }
+
+function formatDurationUntilExpiry( lastUpdatedAt: number ) {
+	const MAX_AGE_IN_DAYS = 7;
+	const now = new Date();
+	const endDate = addDays( lastUpdatedAt, MAX_AGE_IN_DAYS );
+	const difference = endDate.getTime() - now.getTime();
+	let format: DurationUnit[] = [ 'days', 'hours' ];
+
+	if ( difference < HOUR_MS ) {
+		format = [ 'minutes' ];
+	} else if ( difference < DAY_MS ) {
+		format = [ 'hours', 'minutes' ];
+	}
+
+	if ( endDate < now ) {
+		return __( 'Expired' );
+	}
+
+	return formatDuration(
+		intervalToDuration( {
+			start: now,
+			end: addHours( endDate, 1 ),
+		} ),
+		{
+			format,
+			delimiter: ', ',
+		}
+	);
+}
+
+function getColumnWidths( widthFactors: number[] ) {
+	const padding = widthFactors.length * 2;
+	const columns = Math.min( process.stdout.columns || 80, 140 ) - padding;
+	return widthFactors.map( ( widthFactor ) => Math.round( widthFactor * columns ) );
+}
+
+export function getSnapshotCliTable( snapshots: Snapshot[] ) {
+	const colWidths = getColumnWidths( [ 0.4, 0.25, 0.175, 0.175 ] );
+	const table = new Table( {
+		head: [ __( 'URL' ), __( 'Site Name' ), __( 'Updated' ), __( 'Expires in' ) ],
+		wordWrap: true,
+		colWidths,
+		style: {
+			head: [],
+			border: [],
+		},
+	} );
+
+	snapshots.forEach( ( snapshot ) => {
+		const durationUntilExpiry = formatDurationUntilExpiry( snapshot.date );
+
+		table.push( [
+			`https://${ snapshot.url }`,
+			snapshot.name,
+			format( snapshot.date, 'yyyy-MM-dd HH:mm' ),
```

Comment:
I also took this opportunity to change the date format to something more technical. `2025-04-28 08:42` seems like a better fit for the CLI than `Apr 28, 2025, 08:42`.

Reply:
I think this is a great idea!